gitops/utils/apps.py

```diff
@@ -44,18 +44,26 @@ def get_app_details(app_name: str, load_secrets: bool = True, exit_if_not_found:
     return app
 
 
-def update_app(app_name: str, **kwargs: object) -> None:
+def update_app(app_name: str, delete_when_empty: bool = True, **kwargs: object) -> None:
     filename = get_apps_directory() / app_name / "deployment.yml"
     with open(filename) as f:
         data = yaml.safe_load(f)
     for k, v in kwargs.items():
         if k not in DEPLOYMENT_ATTRIBUTES:
             print(warning(f"Key '{k}' is not a recognised deployment attribute for {app_name}."))
         if v in [[], {}]:
-            if k in data:
+            if k not in data:
+                pass
+            elif delete_when_empty or data[k]:
```

Comment:
does `data[k]` do anything here if `v` is already confirmed be falsey?

Reply:
Removing `or data[k]` would cause `test_emptying_environment_variable_should_remove_environment` to fail. 

Without it, we will be setting [`data[k] = v`](https://github.com/uptick/gitops/pull/71/files/c91da78858823a049770457e5c9935339c9d1d7e#diff-83782301e034b2eaf54aa524128e53448e7aa308e1847095db278cce4b58f864R61) (where v is `{}` or `[]`), which will be rendered as empty dict/list: `environment: {}`.

Reply:
oh, I see what I missed. We are iterating over kwargs here, not data